Detectors/TOF/calibration/testWorkflow/TOFCalibCollectorWriterSpec.h

```diff
@@ -0,0 +1,126 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_CALIBRATION_TOFCALIB_COLLECTOR_WRITER_H
+#define O2_CALIBRATION_TOFCALIB_COLLECTOR_WRITER_H
+
+/// @file   TOFCalibCollectorWriterSpec.h
+/// @brief  Device to write to tree the information for TOF time slewing calibration.
+
+#include "TOFCalibration/TOFCalibCollector.h"
+#include "DataFormatsTOF/CalibInfoTOFshort.h"
+#include <TTree.h>
+#include <gsl/span>
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace calibration
+{
+
+class TOFCalibCollectorWriter : public o2::framework::Task
+{
+
+  using Geo = o2::tof::Geo;
+
+ public:
+  void createAndOpenFileAndTree()
+  {
+    TString filename = TString::Format("collTOF_%d.root", mCount);
+    LOG(DEBUG) << "opening file " << filename.Data();
+    mfileOut = new TFile(TString::Format("%s", filename.Data()), "RECREATE");
+    mOutputTree = new TTree("treeCollectedCalibInfo", "Tree with TOF calib info for Time Slewing");
+    mOutputTree->Branch(mOutputBranchName.data(), &mPTOFCalibInfoOut);
+  }
+
+  void init(o2::framework::InitContext& ic) final
+  {
+    mCount = 0;
+    createAndOpenFileAndTree();
+  }
+
+  void run(o2::framework::ProcessingContext& pc) final
+  {
+    auto collectedInfo = pc.inputs().get<std::vector<o2::dataformats::CalibInfoTOFshort>>("collectedInfo");
+    //auto collectedInfo = pc.inputs().get<gsl::span<const o2::dataformats::CalibInfoTOFshort>>("collectedInfo");
+    auto entriesPerChannel = pc.inputs().get<std::array<int, Geo::NCHANNELS>>("entriesCh");
```

Comment:
both inputs provide containers of PODs, you don't need to create their copy by getting vector/array, instead use ``gsl/span``, i.e.
````
auto collectedInfo = pc.inputs().get<gsl::span<o2::dataformats::CalibInfoTOFshort>>("collectedInfo");
auto entriesPerChannel = pc.inputs().get<gsl::span<int>("entriesCh");
````
This will avoid any copy and you can use them as a const vector or array

Reply:
The reason why I make a copy is that I need only a sub-vector of the input, and I did not find a way to do so with gsl::span, as you can see from the comment in the code. Do you have a suggestion?

Reply:
you can create a sub-span of the span at no cost:
````
auto subspanInfo = gsl::span<const o2::dataformats::CalibInfoTOFshort>( &collectedInfo[start], nsel );
````

Reply:
I tried, as written in the comment (it is not mentioned explictly that I used the subspan, but I did), but it did not work. I can try again.

Reply:
Now done, I don't know what I was doing earlier that did not work...